### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by ic-768**
- Simplified immediately invoked function expression for legibility
- Refactored repetitive statements into a function call

### CODE REVIEW

File: pyscriptjs/src/components/base.ts [hunk 1]
```diff
@@ -304,16 +302,14 @@ export class PyWidget extends HTMLElement {
...
-        if (this.hasAttribute('name')) {
-            this.name = this.getAttribute('name');
-        }
+        this.addAttributes('src','name','klass');
+    }
 
-        if (this.hasAttribute('klass')) {
-            this.klass = this.getAttribute('klass');
+    addAttributes(...attrs:string[]){
+        for (const each of attrs){
```
Comment (fpliger) [inline pyscriptjs/src/components/base.ts:309]: This is overwriting `this.source` every time this has the `each` attribute (which is true for `name` and `klass`). It should assign `each` as an attribute of `this` and handle renaming differently. Maybe through a mapping (optional) parameter?

**Review by fpliger — CHANGES_REQUESTED** [review]
Thanks @ic-768 good improvement! I think there's a bug though... commented in line

File: pyscriptjs/src/components/base.ts (hunk 1 above)
Comment (ic-768) [reply pyscriptjs/src/components/base.ts:309]: Oops, you're right. Should be fixed with latest commit

**Review by madhur-tandon — APPROVED** [review]

**Review by fpliger — APPROVED** [review]
lgtm, thank you @ic-768 !

**Pull request merged**